p2p/net/swarm/swarm_dial.go

```diff
@@ -533,3 +532,63 @@ func isTCPMultiaddr(a ma.Multiaddr) bool {
 	p := a.Protocols()
 	return len(p) == 2 && (p[0].Name == "ip4" || p[0].Name == "ip6") && p[1].Name == "tcp"
 }
+
+func isDefaultDockerRange(a ma.Multiaddr) bool {
+	parts := strings.Split(a.String(), "/")
+	if len(parts) != 5 {
+		return false
+	}
+
+	if parts[1] == "ip4" && strings.HasPrefix(parts[2], "172.17.") {
+		return true
+	}
+
+	return false
+}
+
+type AddrList []ma.Multiaddr
```

Comment:
it's a good idea but this should really be the work of an estimator, and a smart dialer thing that can maybe learn which types of addresses wont work.